packages/docs-page/server.js

```diff
@@ -95,49 +97,78 @@ async function generateStaticProps({
   const currentPath = params[paramId] ? params[paramId].join('/') : ''
 
   let versions = []
+  let latestVersion = null
 
   // This code path handles versioned docs integration, which is currently gated behind the ENABLE_VERSIONED_DOCS env var
-  if (process.env.ENABLE_VERSIONED_DOCS) {
-    const versionFromPath = getVersionFromPath(params[paramId])
-
-    const currentVersionNormalized = normalizeVersion(currentVersion)
+  if (ENABLE_VERSIONED_DOCS) {
+    // given: v0.5.x (latest), v0.4.x, v0.3.x
+    const [versionFromPath, paramsNoVersion] = stripVersionFromPathParams(
+      params[paramId]
+    )
+    // versionFromPath should realistically only ever be "latest" | "v0.4.x" | "v0.3.x"
+    // It could be v0.5.x if a use navigates directly to it.
 
-    versions = await loadVersionList(product.slug)
+    const versionMetadataList = await fetchVersionMetadataList(productSlug)
```

Comment:
Let's use the cached version as this API response shouldn't change at all during a build!

```suggestion
    const versionMetadataList = await cachedFetchVersionMetadataList(productSlug)
```